test/pytest/receiver/test_asapo_transfer.py

```diff
@@ -157,6 +157,23 @@ def test_path_parsing(asapo_transfer, file_list, hidra_metadata):
         assert data_source == 'test001'
 
 
+def test_path_parse_offset(asapo_transfer):
+    ragexp_template = "current/raw/(?P<scan_id>.*)_(?P<file_idx_offset>.*)_(?P<file_idx_in_scan>.*).h5"
+    asapo_transfer.asapo_worker.file_regex = ragexp_template
+
+    file_path = "current/raw/data/tilt_27_048_041_data_070421_000001.h5"
+    data_source, stream, file_idx = asapo_transfer.asapo_worker._parse_file_name(file_path)
+    assert file_idx == 70422
+
+    file_path = "current/raw/data/tilt_27_048_041_data_070421.h5"
+    data_source, stream, file_idx = asapo_transfer.asapo_worker._parse_file_name(file_path)
+    assert file_idx == 70421
+
+    with pytest.raises(ValueError):
+        file_path = "current/raw/data/tilt_27_048_041_data_070421_foo.h5"
+        data_source, stream, file_idx = asapo_transfer.asapo_worker._parse_file_name(file_path)
```

Comment:
This has nothing to do with the offset and should be a separate test.

If not too difficult, it would be better to (also) test the final result of throwing an exception, i.e., is the file still ingested to asapo and if yes, to which data source, stream, and index, or will the worker crash, and if yes, will it restart.